src/components/data-management/ProjectMenu.jsx

```diff
@@ -54,6 +60,64 @@ const ProjectMenu = (props) => {
     });
     return canLaunch;
   };
+
+  const getProjectHash = (project) => {
+    const samplesToHash = project.samples.map((sampleUuid) => ({
+      name: samples[sampleUuid].name,
+      metadata: samples[sampleUuid].metadata,
+    }));
+
+    return hash.MD5(samplesToHash);
+  };
+
+  // Initialize hash for each project
+  useEffect(() => {
+    if (!activeProjectUuid) return;
+
+    if (!initialProjectState.current[activeProjectUuid]) {
+      initialProjectState.current[activeProjectUuid] = getProjectHash(activeProject);
+    }
+
+    const reasons = [];
+
+    const experimentId = activeProject.experiments[0];
+    const gem2sStatus = backendStatus[experimentId]?.status.gem2s?.status;
+
+    const gem2sNotSuccessful = ![pipelineStatus.SUCCEEDED, pipelineStatus.RUNNING].includes(gem2sStatus);
+    const ProjectHashNotEqual = initialProjectState.current[activeProjectUuid] !== getProjectHash(activeProject);
+
+    if (gem2sNotSuccessful) reasons.push('data has not been processed sucessfully');
+    if (ProjectHashNotEqual) reasons.push('the project has been modified');
+
+    setWillGem2sRerun({
+      rerun: gem2sNotSuccessful || ProjectHashNotEqual,
+      reasons,
+    });
+  }, [backendStatus, activeProjectUuid, samples]);
+
+  const reingestIndicator = (rerun) => (
+    <div style={{
+      width: '0.75rem',
+      height: '0.75rem',
+      marginLeft: '0.6rem',
+      display: 'inline-block',
+      borderRadius: '50%',
+      verticalAlign: 'middle',
+      backgroundColor: rerun ? 'tomato' : 'lightgreen',
+    }}
+    />
+  );
+
+  const launchTooltipMessage = () => {
+    if (activeProject?.samples.length > 0 && !canLaunchAnalysis()) {
+      return `Ensure all samples are uploaded and all metadata are inserted (no ${DEFAULT_NA})`;
+    }
+
+    if (willGem2sRerun.rerun) {
```

Comment:
Can we create and use `gem2sNotSuccessful` and `ProjectHashNotEqual` in this function directly?